akka-http-core/src/main/scala/akka/http/impl/engine/client/OutgoingConnectionBlueprint.scala

```diff
@@ -169,21 +169,17 @@ private[http] object OutgoingConnectionBlueprint {
       private def entitySubstreamStarted = entitySource ne null
       private def idle = this
       private var completionDeferred = false
+      private var completeOnMessageEnd = false
```

Comment:
Thought about that too but wasn't sure if `completionDeferred ` is guaranteed to be set after completeOnMessageEnd. `completionDeferred  |= closeRequested ` should work though.